packages/interface-libp2p/src/index.ts

```diff
@@ -87,7 +88,14 @@ export interface LookupFunction {
 /**
  * Libp2p nodes implement this interface.
  */
-export interface Libp2p extends Startable, EventEmitter<Libp2pEvents> {
+export interface Libp2p<
+  FindPeerProgressEvents extends ProgressEvent = ProgressEvent,
+  GetClosestPeersProgressEvents extends ProgressEvent = ProgressEvent,
+  ProvideProgressEvents extends ProgressEvent = ProgressEvent,
+  FindProvidersProgressEvents extends ProgressEvent = ProgressEvent,
+  PutProgressEvents extends ProgressEvent = ProgressEvent,
+  GetProgressEvents extends ProgressEvent = ProgressEvent
+> extends Startable, EventEmitter<Libp2pEvents> {
```

Comment:
you should add generics to the `extends ProgressEvent` in order to appropriately populate progressEvent or else users won't be able to customize their data since it's always going to be set to `ProgressEvent<any, unknown>`

e.g. 

```suggestion
export interface Libp2p<
  FindPeerT extends string = any, FindPeerD = unknown,
  GetClosestPeersT extends string = any, GetClosestPeersD = unknown,
  ProvideT extends string = any, ProvideD = unknown,
  FindProvidersT extends string = any, FindProvidersD = unknown,
  PutT extends string = any, PutD = unknown,
  GetT extends string = any, GetD = unknown,
> extends Startable, EventEmitter<Libp2pEvents> {
```

this can obviously be improved, but without piping the generics through, you will lock types to any and unknown, and aren't getting much benefit